non-critical-infra/packages/encrypt-email-address/encrypt-email-address.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+import re
+import click
+import subprocess
+from pathlib import Path
+from textwrap import dedent
+from textwrap import indent
+
+
+@click.command(
+    help=dedent("""\
+        Encrypt an email address (or email addresses) for inclusion in a mailing list.
+
+        See `non-critical-infra/modules/mailserver/mailing-lists.nix` for
+        documentation about how to use this.
+    """)
+)
+@click.argument("id")
+@click.argument("email")
+@click.option("--force/--no-force", "-f/ ", default=False)
+def main(id: str, email: str, force: bool):
+    # Feel free to make the regex less restrictive if you need to.
+    id_re = re.compile("[A-Za-z0-9-]+")
+    if not id_re.fullmatch(id):
+        raise click.ClickException(
+            f"Given ID: {id!r} is invalid. Must match regex: {id_re.pattern}"
+        )
+
+    # Make sure we aren't being given a text file that happens to have a newline at the end.
+    if "\n" in email:
+        raise click.ClickException(
+            "Email address must not not contain a newline character"
+        )
+
+    if Path.cwd().name != "non-critical-infra":
+        raise click.ClickException(
+            "You must run this command inside the `non-critical-infra/` folder."
+        )
```

Comment:
Just changedir to the correct directory?

Reply:
You can calculate the script directory using `Path(__name__).parent` and than compute the path you want to patch relative to that.

Reply:
I've implemented 1) by searching upwards for a `.git/config` directory. It was pretty simple to do: https://github.com/NixOS/infra/pull/495/commits/240fff84baebf4372d72bd4967ce30bb27ae1638. Now you can run this command from anywhere in the project =)

I'm going to mark this resolved.